src/java.base/share/classes/java/util/concurrent/locks/AbstractQueuedLongSynchronizer.java

```diff
@@ -1185,6 +1232,26 @@ private void unlinkCancelledWaiters(ConditionNode node) {
             }
         }
 
+        /**
+         * Constructs objects needed for condition wait. On OOME,
+         * releases lock, sleeps, reacquires, and returns null.
+         */
+        private ConditionNode newConditionNode() {
+            long savedState;
+            if (tryInitializeHead() != null) {
+                try {
+                    return new ConditionNode();
+                } catch (OutOfMemoryError oome) {
+                }
+            }
+            // fall through if encountered OutOfMemoryError
+            if (!isHeldExclusively() || !release(savedState = getState()))
+                throw new IllegalMonitorStateException();
```

Comment:
I still don't see how an OOME can possibly indicate we might somehow be in a state where we need to throw IMSE!